src/vivarium/framework/randomness.py

```diff
@@ -84,19 +84,20 @@ def update(self, new_keys: Index):
         if not self._map.index.intersection(new_keys).empty:
             raise KeyError("Non-unique keys in index.")
 
-        mapping_update = self.hash_(new_keys)
-        if self._map.empty:
-            self._map = mapping_update.drop_duplicates()
-        else:
-            self._map = pd.concat([self._map, mapping_update]).drop_duplicates()
+        if not new_keys.empty:
```

Comment:
I think it's probably more readable to do 
```
if new_keys.empty:
    return  # Nothing to do
elif not self._map.index.intersection(new_keys).empty:
    raise KeyError("Non-unique keys in index")

...old method body...
```
[`Flat is better than nested`](https://peps.python.org/pep-0020/)